includes/Core/User_Input/User_Specific_Questions.php

```diff
@@ -0,0 +1,113 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\User_Input\User_Specific_Questions
+ *
+ * @package   Google\Site_Kit\Core\User_Input
+ * @copyright 2022 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\User_Input;
+
+use Closure;
+use Google\Site_Kit\Core\Storage\User_Setting;
+
+/**
+ * Class for handling the user specific settings in User Input.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class User_Specific_Questions extends User_Setting {
+
+	/**
+	 * The user option name for this setting.
+	 */
+	const OPTION = 'googlesitekit_user_input_settings';
+
+	/**
+	 * The scope for which the questions are handled by this class.
+	 */
+	const SCOPE = 'user';
+
+	/**
+	 * Gets the expected value type.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return string The type name.
+	 */
+	protected function get_type() {
+		return 'array';
+	}
+
+	/**
+	 * Gets the default value.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array The default value.
+	 */
+	protected function get_default() {
+		return array();
+	}
+
+	/**
+	 * Gets the callback for sanitizing the setting's value before saving.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return Closure
+	 */
+	protected function get_sanitize_callback() {
+		$questions = array_filter(
+			User_Input::get_questions(),
+			function ( $question ) {
+				return static::SCOPE === $question['scope'];
+			}
+		);
+
+		return function ( $settings ) use ( $questions ) {
+			if ( ! is_array( $settings ) ) {
+				return $this->get();
+			}
+
+			$results = array();
+
+			foreach ( $settings as $setting_key => $setting_values ) {
+				// Ensure all the data is valid.
+				if (
+					! in_array( $setting_key, array_keys( $questions ), true ) ||
+					! is_array( $setting_values ) ||
+					static::SCOPE !== $setting_values['scope'] ||
+					! is_array( $setting_values['values'] )
+				) {
+					continue;
+				}
+
+				$valid_values          = array();
+				$valid_values['scope'] = $setting_values['scope'];
+
+				$valid_answers = array();
+
+				// Verify that each answer value is a string.
+				foreach ( $setting_values['values'] as $answer ) {
+					if ( is_string( $answer ) ) {
```

Comment:
Why specifically must the answers be strings? Maybe `is_scalar` would be a more appropriate condition? I suppose they're all strings today, but it might be odd if we chose to accept a number in the future and it were to filter that out because of this strict condition. Just a thought. What do you think?

Reply:
Valid call. Updated to use `is_scalar()`, thank you!